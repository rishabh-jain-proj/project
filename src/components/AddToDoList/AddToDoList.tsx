import React, { useState, useRef } from 'react';

interface Props {
    addText: (todoText: string) => void
}

export const AddToDoList: React.FC<Props> = ({ addText }) => {
    const [todoText, setTodoText] = useState<string>('');
    const todoRef = useRef<HTMLInputElement>(null);
    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const newTodoText: string = e.target.value;
        setTodoText(newTodoText);
    }

    const submitText = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        if (todoText.trim()) {
            addText(todoText.trim());
        } else{
            alert('Empty Spaces Aren\'t Allowed');
        }
        setTodoText('');
    }

    return (
        <form onSubmit={(e) => submitText(e)} className="col s12">
            <div className="row">
                <div className="input-field col s6">
                    <input value={todoText} onChange={(e) => handleChange(e)} ref={todoRef} placeholder="Enter Todos" id="todos_text" type="text" />
                </div>
                <div   title={`${todoText ? 'Add Todos' : 'No Todos To Add'}`}  style={{padding: '1rem 0rem'}} className=" col s6">
                    <button id="submit" className={`waves-effect waves-light btn ${todoText ? '' : 'disabled'}`}>Add</button>
                </div>
            </div>
        </form>
    )
}
