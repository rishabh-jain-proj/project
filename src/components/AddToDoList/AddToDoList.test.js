import React from 'react';
import ReactDom from 'react-dom';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from "enzyme-adapter-react-16";
import { AddToDoList } from './AddToDoList';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";


Enzyme.configure({
    adapter: new EnzymeAdapter(),
    disableLifecycleMethods: true
});

describe('AddToDoList component testing', () => {
    afterEach(cleanup);
    let wrapper;

    it("should render without crashing", () => {
        const div = document.createElement("div");
        ReactDom.render(<AddToDoList addText={jest.fn()} />, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it("should render AddToDoList correctly", () => {
        wrapper = render(<AddToDoList addText={jest.fn()} />)
        expect(wrapper).toMatchSnapshot();
    });

    it("should Render Add Button with disabled state", () => {
        wrapper = mount(<AddToDoList addText={jest.fn()} />);
        let button = wrapper.find("#submit")
        expect(button.hostNodes().render().text())
            .toEqual('Add')
        button = wrapper.find(".disabled")
        expect(button).toHaveLength(1)
    }) 
});