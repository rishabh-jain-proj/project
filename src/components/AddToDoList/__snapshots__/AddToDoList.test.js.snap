// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToDoList component testing should render AddToDoList correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="col s12"
      >
        <div
          class="row"
        >
          <div
            class="input-field col s6"
          >
            <input
              id="todos_text"
              placeholder="Enter Todos"
              type="text"
              value=""
            />
          </div>
          <div
            class=" col s6"
            style="padding: 1rem 0rem;"
            title="No Todos To Add"
          >
            <button
              class="waves-effect waves-light btn disabled"
              id="submit"
            >
              Add
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="col s12"
    >
      <div
        class="row"
      >
        <div
          class="input-field col s6"
        >
          <input
            id="todos_text"
            placeholder="Enter Todos"
            type="text"
            value=""
          />
        </div>
        <div
          class=" col s6"
          style="padding: 1rem 0rem;"
          title="No Todos To Add"
        >
          <button
            class="waves-effect waves-light btn disabled"
            id="submit"
          >
            Add
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
